src/components/Exchange/SwapBox.js

```diff
@@ -1642,10 +1643,14 @@ export default function SwapBox(props) {
   };
 
   const onSwapOptionChange = (opt) => {
+    // Keep same fromToken when toggling swap option
+    const _swapOption = swapOption;
+    const prevTokenAddress = tokenSelection[_swapOption].from;
+    const updatedTokenSelection = JSON.parse(JSON.stringify(tokenSelection));
+    updatedTokenSelection[opt].from = prevTokenAddress;
+    setTokenSelection(updatedTokenSelection);
```

Comment:
or what does setTokenSelection do?

Reply:
tokenSelection stores the previously selected tokens for each swap option that the user made. So it overrides the next swap option's token to be the current token. E.g. User is on Long with fromToken changed to USDC and they change to Short, setTokenSelection sets the fromToken in Short to USDC as well

Reply:
ok, ty